### PR TITLE
feat: [CDS-52630]: Added links audit logs for different entity

### PR DESCRIPTION
**Pull request by kapilkumawat86**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/RouteDestinations.tsx [hunk 1]
```diff
@@ -212,60 +212,139 @@ LandingDashboardFactory.registerModuleDashboardHandler(ModuleName.CD, {
...
+  }
+})
+
+AuditTrailFactory.registerResourceHandler('ENVIRONMENT_GROUP', {
+  moduleIcon: {
+    name: 'cd-main'
+  },
+  moduleLabel: cdLabel,
+  resourceLabel: 'common.environmentGroup.label',
+  resourceUrl: (environment: ResourceDTO, resourceScope: ResourceScope, module?: Module) => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/RouteDestinations.tsx:285]:
```suggestion
  resourceUrl: (environmentGroup: ResourceDTO, resourceScope: ResourceScope, module?: Module) => {
```

File: src/modules/70-pipeline/RouteDestinations.tsx [hunk 2]
```diff
@@ -212,60 +212,139 @@ LandingDashboardFactory.registerModuleDashboardHandler(ModuleName.CD, {
...
     const { accountIdentifier, orgIdentifier, projectIdentifier } = resourceScope
-    if (module && orgIdentifier && projectIdentifier) {
-      return routes.toServices({
+    if (environment.identifier && orgIdentifier && projectIdentifier) {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/RouteDestinations.tsx:287]:
```suggestion
    if (environmentGroup.identifier && orgIdentifier && projectIdentifier) {
```

File: src/modules/70-pipeline/RouteDestinations.tsx [hunk 3]
```diff
@@ -212,60 +212,139 @@ LandingDashboardFactory.registerModuleDashboardHandler(ModuleName.CD, {
...
+      return routes.toEnvironmentGroupDetails({
         module,
         orgIdentifier,
         projectIdentifier,
-        accountId: accountIdentifier
+        accountId: accountIdentifier,
+        environmentGroupIdentifier: environment.identifier
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/RouteDestinations.tsx:293]:
```suggestion
        environmentGroupIdentifier: environmentGroup.identifier
```

File: src/modules/72-templates-library/RouteDestinations.tsx [hunk 4]
```diff
@@ -54,20 +54,23 @@ RbacFactory.registerResourceTypeHandler(ResourceType.TEMPLATE, {
  * Register for Audit Trail
  * */
 const cdLabel = 'common.purpose.cd.continuous'
-AuditTrailFactory.registerResourceHandler(ResourceType.TEMPLATE, {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-templates-library/RouteDestinations.tsx:57]: Do we not want to rely on this ResourceType from rbac anymore? Will we have a different resource type for audit trail later?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**